src/Couchbase.TravelSample/Program.cs

```diff
@@ -103,12 +336,995 @@
+                {
+                    return Results.NotFound();
+                }
+            }
+            else
+            {
+                return Results.Problem("Scope not found");
+            }
+        }
+        catch (DocumentNotFoundException)
+        {
+            Results.NotFound();
+        }
+        catch (Exception ex)
+        {
+            return Results.Problem(ex.Message);
+        }
+
+        return Results.NotFound();
+    })
+    .WithTags("Airport")
+    .WithOpenApi(operation => new(operation)
+    {
+        Description = "Delete Airport with specified ID.\n\nThis provides an example of using Key Value operations in Couchbase to delete a document with specified ID.",
+        Parameters = new List<OpenApiParameter>
+        {
+            new OpenApiParameter
+            {
+                Name = "id",
+                In = ParameterLocation.Path,
+                Description = "Airport ID like airport_1273",
+                Required = true
+            }
+        },
+        Responses = new OpenApiResponses
+        {
+            ["204"] = new OpenApiResponse
+            {
+                Description = "Airport Deleted"
+            },
+            ["404"] = new OpenApiResponse
+            {
+                Description = "Airport ID not found"
+            },
+            ["500"] = new OpenApiResponse
+            {
+                Description = "Unexpected Error"
+            }
+        }
+        
+    });
+
+app.MapGet("/api/v1/airline/list", async (string? country, int? limit, int? offset) =>
+    {
+        try
+        {
+            if (inventoryScope is not null)
+            {
+                // Set default values for limit and offset if not provided by the user
+                limit ??= 10;
+                offset ??= 0;
+
+                var query = string.IsNullOrEmpty(country)
+                    ? $@"SELECT airline.callsign,
+                            airline.country,
+                            airline.iata,
+                            airline.icao,
+                            airline.name
+                            FROM airline AS airline
+                            ORDER BY airline.name
+                            LIMIT $limit
+                            OFFSET $offset"
+                    : $@"SELECT airline.callsign,
+                            airline.country,
+                            airline.iata,
+                            airline.icao,
+                            airline.name
+                            FROM airline AS airline
+                            WHERE lower(airline.country) = $country
+                            ORDER BY airline.name
+                            LIMIT $limit
+                            OFFSET $offset";
+
+                //setup parameters
+                var queryParameters = new Couchbase.Query.QueryOptions();
+                queryParameters.Parameter("country", string.IsNullOrEmpty(country) ? "" : country.ToLower());
```

Comment:
Is this code really going to work with the query you have set above?  Have you tested this to validate when country is in fact empty this doesn't throw an exception because the parameters is missing?  Why even create the parameter - you don't add it in the query so why isn't this line in the same if statement to see if it's not null.  All this code can be rearranged to make a single if check and then make the custom query and add it to the query parameter options.

Reply:
Yes, I've tested it and it doesn't show any exception while country is empty. I've refactored the code as you have told.